server/sonar-server/src/main/java/org/sonar/server/activity/index/ActivityIndex.java

```diff
@@ -67,35 +65,33 @@ public SearchResponse doSearch(ActivityQuery query, SearchOptions options) {
     requestBuilder.setSize(options.getLimit());
     requestBuilder.addSort(ActivityIndexDefinition.FIELD_CREATED_AT, SortOrder.DESC);
 
-    AndFilterBuilder filter = FilterBuilders.andFilter();
+    AndQueryBuilder filter = QueryBuilders.andQuery();
     if (!query.getTypes().isEmpty()) {
-      OrFilterBuilder typeFilter = FilterBuilders.orFilter();
+      OrQueryBuilder typeQuery = QueryBuilders.orQuery();
       for (String type : query.getTypes()) {
-        typeFilter.add(FilterBuilders.termFilter(ActivityIndexDefinition.FIELD_TYPE, type));
+        typeQuery.add(QueryBuilders.termQuery(ActivityIndexDefinition.FIELD_TYPE, type));
       }
-      filter.add(typeFilter);
+      filter.add(typeQuery);
     }
 
     if (!query.getDataOrFilters().isEmpty()) {
       for (Map.Entry<String, Object> entry : query.getDataOrFilters().entrySet()) {
-        OrFilterBuilder orFilter = FilterBuilders.orFilter();
-        orFilter.add(FilterBuilders.nestedFilter(ActivityIndexDefinition.FIELD_DETAILS,
-          FilterBuilders.termFilter(ActivityIndexDefinition.FIELD_DETAILS + "." + entry.getKey(), entry.getValue())));
-        filter.add(orFilter);
+        OrQueryBuilder orQuery = QueryBuilders.orQuery();
+        orQuery.add(QueryBuilders.nestedQuery(ActivityIndexDefinition.FIELD_DETAILS,
+          QueryBuilders.termQuery(ActivityIndexDefinition.FIELD_DETAILS + "." + entry.getKey(), entry.getValue())));
+        filter.add(orQuery);
       }
     }
 
     Date since = query.getSince();
     if (since != null) {
-      filter.add(FilterBuilders.rangeFilter(ActivityIndexDefinition.FIELD_CREATED_AT)
-        .gt(since)
-        .cache(false));
+      filter.add(QueryBuilders.rangeQuery(ActivityIndexDefinition.FIELD_CREATED_AT)
+        .gt(since));
     }
     Date to = query.getTo();
     if (to != null) {
-      filter.add(FilterBuilders.rangeFilter(ActivityIndexDefinition.FIELD_CREATED_AT)
-        .lt(to)
-        .cache(false));
+      filter.add(QueryBuilders.rangeQuery(ActivityIndexDefinition.FIELD_CREATED_AT)
+        .lt(to));
     }
 
     requestBuilder.setQuery(QueryBuilders.filteredQuery(QueryBuilders.matchAllQuery(), filter));
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove this use of "filteredQuery"; it is deprecated. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)